src/lib/Microsoft.Health.Extensions.Fhir.R4/Search/SearchExtensions.cs

```diff
@@ -197,5 +198,51 @@ public static string ToSearchQueryParameter(this Hl7.Fhir.Model.Identifier ident
 
             return $"identifier={identifier.ToSearchToken()}";
         }
+
+        public static string ToExpandedSearchQueryParameters(this Hl7.Fhir.Model.Identifier identifier)
+        {
+            EnsureArg.IsNotNull(identifier?.Value, nameof(identifier.Value));
+
+            string[] components = identifier.Value?.Split('.');
+
+            // Check if the identifier contains at least 5 components separated by '.', e.g. 'patientId.deviceId.typeName.startTime.endTime'.
+            if (components.Length >= 5)
+            {
+                // Make sure the start component can be converted to an ISO 8601 time string.
+                if (TryGetFormattedDateString(components[components.Count() - 2], out string start))
+                {
+                    // Make sure the end component can be converted to an ISO 8601 time string.
+                    if (TryGetFormattedDateString(components[components.Count() - 1], out string end))
+                    {
+                        // Merge any componets the might make up the typeName.
+                        // This will happen if the type name contains one or more '.'.
+                        string[] codeComponents = components.Skip(2).Take(components.Length - 4).ToArray();
+                        string code = string.Join('.', codeComponents);
+
+                        // Return the expanded search string.
+                        return $"subject={components[0]}&device={components[1]}&code={code}&date=ge{start}&date=le{end}";
+                    }
+                }
+            }
+
+            // Default to searching with the identifier.

```

Comment:
nit: Should we log identifiers that don't fit the 'at least 5 components' pattern as an information, or would that be unnecessary as it is a normal scenario too.

Reply:
We could, but this condition is "normal" and can be common for some customers. if the Observations are referencing correlations the identifier uses the correlation id instead of start and end dates.